src/Workspaces/Remote/ServiceHub/Shared/RemoteEndPoint.cs

```diff
@@ -142,30 +181,153 @@ public async Task<T> InvokeAsync<T>(string targetName, IReadOnlyList<object> arg
 
             try
```

Comment:
ok.  so this looks like massive duplciation, jus tto support the case of Invoke not having a return type.  But htat can be simulated just by having that version call this version witha  dummy arg (like `InvokeAsync<object>`) right?

Then w wouldnt' need so much copy/paste.

Reply:
I don't think the duplication is so massive now. We're gonna likely remove all these once we move to `IServiceBroker` anyways, so I'll keep it as is for now.